### ISSUE TITLE
Content-Type constraint validation

### ISSUE THREAD

**Issue by mcweba** [swisspush/gateleen#340]
### Introduction
The current security related feature set of gateleen provides the ability to grant or restrict access to resources (URLs) based on user roles (ACLs). Also the method (GET, PUT, POST, etc.) which is allowed can be defined.

However, there's no option to restrict allowed file types for URLs.

### Feature Request
* Ability to provide granted file types based on URLs
* Default file types which are always allowed should be configurable
* File type definitions from requests come from the `Content-Type` request header
* Actual file type validation in the payload is not required. This should be handled elsewhere (Webserver, Proxy, etc.)

### Approach
Provide a new handler called something like `ContentTypeConstraintHandler`. This handler takes a list
of default content types which are allways (for all URLs) allowed. The handler also takes a resource path pointing
to the configuration resource for dynamic content type constraint configuration.

This resource could look something like this:
```json
{
  "/gateleen/contacts/zips/(.*)": {
    "description": "contacts are uploaded as zip files",
    "allowedTypes": ["application/zip"]
  },
  "/gateleen/contacts/storage/(.*)": {
    "description": "contact storage contains images and videos",
    "allowedTypes": ["image/png", "image/bmp", "video/mp4"]
  }
}
```
When a request with unwanted content type is found, it will be responded with a `415 Unsupported Media Type` status code.

**Additional considerations**

Should the content type entries in the configuration resource be matched like an _equalsIgnoreCase_ or should they
be regex patterns like `image/.*` ?

Also should we ignore additional specification entries like `application/xhtml+xml` and treat them like `application/xml` for example?
This would result in stripping everything between the `/` and the `+`.

Taking the example above: What if we want to explicitly prohibit `xhtml` files, but allow other `xml` files? Using a regex pattern,
this could be solved by `application/((?!xhtml).*)\+xml`

According to [www.iana.org](https://www.iana.org/assignments/media-types/media-types.xhtml), there are potentially lots of app specific content types, which should be handled correctly.

@lbovet what do you think?

**Comment by lbovet** [comment]
1. Yes, good idea. we could match with regexps like `image/.*`
2. Maybe it brings a bit of unwanted complexity. Users can match them with `application/.*+?xml` thanks to the above feature.

**Comment by mcweba** [comment]
Released in [v1.1.59](https://github.com/swisspush/gateleen/releases/tag/v1.1.59)
